### ISSUE TITLE
TypeError: Cannot read property 'full' of undefined / Worker error Error: No code sections found

### ISSUE THREAD

**Issue by jodriscoll** [FormidableLabs/webpack-dashboard#231]
> Please provide a description of the bug / issue, and provide the details below:
The `webpack-dashboard` gets through the initial configuration process (such as the console output and bundling process), then appears to fail shortly after.

Issue listed within the "Problems" section as
```bash
1: index.bundle.js

No duplicate files!
Unable to diagnose possible version skews
```

NPM error
`TypeError: Cannot read property 'full' of undefined`

Video demonstration 
- http://recordit.co/3DPTBuhP9z (sorry for the large resolution!)

====================================================================

###### If the issue is visual, please provide screenshots here
Dashboard screen: 
* https://image.prntscr.com/image/HnXjxUGLQOqVIyICN6_LuA.png

NPM Error: 
* https://image.prntscr.com/image/kz_bWYPnQjaHqxdcLPPIAg.png

====================================================================

###### Steps to reproduce the problem
1. Installed `webpack-dashboard`
2. Defined `const`
3. Added to the `plugins: [ ]`
4. Added to the `package.json { ... start: }` script
5. Executed `npm start`
6. `webpack-dashboard` was cranking away until it proceeded past the `progress` bar completion
7. Error introduced that is being reported

====================================================================

###### Please provide a gist of relevant files

1. package.json (specifically the script you are using to start the dashboard)
    - https://gist.github.com/jodriscoll/f49539145957855ca1229033bb217668
2. webpack.config.js
    - https://gist.github.com/jodriscoll/9cb754fa927a3a7578b643652cf26bb0
3. index.js (Your express based dev server, if applicable)
    - N/A

====================================================================

###### More Details

- What operating system are you on?
  * macOS High Sierra (`10.13.2`)
- What terminal application are you using?
  * iTerm2 paired with fish-shell
- What version of webpack-dashboard are you using?
  * `^1.1.0`
- What is the output of running `echo $TERM`?
  * `xterm-256color`

Additional observation:
If I place the plugin at the bottom of the `plugins` array, it doesn't error out, but instead spits out multiple instances of `Worker error Error: No code sections found` (_shown towards the end of the video demonstration_)

**Comment by jodriscoll** [comment]
Regarding the "additional observation" section at the bottom of my issue, I located https://github.com/FormidableLabs/webpack-dashboard/issues/189, however, I'm not sure it is relative to my circumstance?

**Comment by ryan-roemer** [comment]
@jodriscoll -- Can you get me a full stack trace? It's very, very likely related to a change I just published today: https://github.com/FormidableLabs/webpack-dashboard/pull/230

Also, if possible, can you create a new minimal public repository with install + error reproduction steps so I can dive into the failure directly? Thanks!

**Comment by ryan-roemer** [comment]
Ah found stack trace. Investigating...

**Comment by jodriscoll** [comment]
Fantastic! Let me know if you still need me to prop up a clone-able instance. 👍

**Comment by ryan-roemer** [comment]
@jodriscoll -- Can you try out https://github.com/FormidableLabs/webpack-dashboard/compare/bug/stats-undefined and let me know if that fixes things?

**Comment by jodriscoll** [comment]
Good news, no more `TypeError: Cannot read property 'full' of undefined` with that recommendation!
  - http://recordit.co/8DlQ1x5Fg6

However, I'm still experiencing the second part of the issue:
  - (initial compile) https://image.prntscr.com/image/KKFVMoXdSeWO4Y5EOqQqSg.png
  - (change occurrence, re-bundle) https://image.prntscr.com/image/subF1Et7QqyxTCvsymcsDQ.png

**Comment by ryan-roemer** [comment]
OK, I'm going to PR up the fix for the first thing now. Can you get me a minimal repository that hits the error with install + error reproduction instructions? Thanks!

**Comment by jodriscoll** [comment]
Invitation sent! Hopefully that is enough to help further diagnose.

**Comment by ryan-roemer** [comment]
@jodriscoll -- Thanks! Cloning and digging in. Will update if I run into kinks.

**Comment by ryan-roemer** [comment]
@jodriscoll -- I'm getting some build errors like:

```
ERROR in ./src/views/Widgets/Widget03.js                                   
Module build failed: SyntaxError: Unexpected token (18:42)                 
                                                                           
  16 | class Widget03 extends Component                                    
  17 |   render()                                                          
> 18 |     const className, cssModule, dataBox, ...attributes = this.props;
     |                                           ^                         
  19 |                                                                     
  20 |     // demo purposes only                                           
  21 |     const data = dataBox();                                         
                                                                           
 @ ./src/views/Widgets/Widgets.js 23:15-36                                 
 @ ./src/containers/Dashboard/Dashboard.js                                 
 @ ./src/index.js                                                          
```

when running `npm start`. Is that what happens on your end?

**Comment by ryan-roemer** [comment]
@jodriscoll -- Your code sections aren't being detected because you need to turn off minification in dev for use with the dashboard. (The dashboard can't infer the code sections when minified).

This worked for me:

```diff
diff --git a/webpack.config.js b/webpack.config.js
index b450cc4..ca7bd94 100755
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -29,7 +29,7 @@ module.exports = (env = {}) => {
     devServer: {
       contentBase: BUILD_DIR,
       // port: 9001,
-      compress: true,
+      compress: false,
       hot: true,
       open: true
     },
@@ -101,7 +101,7 @@ module.exports = (env = {}) => {
     },
     plugins: [
       new webpack.HotModuleReplacementPlugin(),
-      new webpack.optimize.UglifyJsPlugin({sourceMap: true}),
+      //new webpack.optimize.UglifyJsPlugin({sourceMap: true}),
       new webpack.NamedModulesPlugin(),
       extractCSS,
       extractSCSS,
```

(And you can do something more sophisticated like having dev vs. prod configs and/or using environment variables to control).

With those changes, `npm start` now looks like:

```
┌─Log───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐┌─Status────────────────────────────────────────┐
│                                                                                                                                                   ││                                               │
│ Project is running at http://localhost:8080/                                                                                                      ││                                               │
│                                                                                                                                                   ││ Failed                                        │
│ webpack output is served from /                                                                                                                   ││                                               │
│ Content not from webpack is served from /Users/rye/scm/os/jodriscoll_webkit-dashboard_issue231/build                                              │└───────────────────────────────────────────────┘
│                                                                                                                                                   │┌─Operation─────────────────────────────────────┐
│ webpack: wait until bundle finished: /                                                                                                            ││                                               │
│                                                                                                                                                   ││                                               │
│ webpack: wait until bundle finished: /index.bundle.js                                                                                             ││ idle (13s)                                    │
│ webpack: Failed to compile.                                                                                                                       ││                                               │
│                                                                                                                                                   │└───────────────────────────────────────────────┘
│ Hash: a846c3ce5a66f92794c0                                                                                                                        │┌─Progress──────────────────────────────────────┐
│ Version: webpack 3.8.1                                                                                                                            ││                                               │
│ Time: 12924ms                                                                                                                                     ││                                               │
│                                                              Asset       Size  Chunks                    Chunk Names                              ││ 0%                                            │
│                                            img/avatars/jcorrao.jpg     7.3 kB          [emitted]                                                  ││                                               │
│   ./fonts/fontawesome-webfont.674f50d287a8c48dc19ba404d20fe713.eot     166 kB          [emitted]                                                  │└───────────────────────────────────────────────┘
│                                                                                                                                                   │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌──Modules────────────────────────────────────────────────────────────────────────────────────────┐┌──Assets─────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                 ││                                                                                                 │
│   1:index.bundle.js                                                                             ││                                                                                                 │
│                                                                                                 ││  Name                                                                    Size                   │
│  Name                                      Size (min+gz)             Percentage                 ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││  img/avatars/jcorrao.jpg                                                 7.13 KB                │
│  moment@2.18.1                             105.59 KB                 24.2%                      ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││  ./fonts/fontawesome-webfont.674f50d287a8c48dc19ba404d20fe713.eot        161.86 KB              │
│  react-dom@16.1.1                          78.59 KB                  18.0%                      ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│  chart.js                                  52.98 KB                  12.1%                      ││                                                                                                 │
│                                                                                                 │└─────────────────────────────────────────────────────────────────────────────────────────────────┘
│                                                                                                 │┌─Problems────────────────────────────────────────────────────────────────────────────────────────┐
│  views                                     30.45 KB                  6.97%                      ││                                                                                                 │
│                                                                                                 ││  1:index.bundle.js                                                                              │
│                                                                                                 ││                                                                                                 │
│  sockjs-client                             19.98 KB                  4.57%                      ││  No problems detected!                                                                          │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│  lodash@4.17.4                             19.14 KB                  4.38%                      ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│  reactstrap                                17.22 KB                  3.94%                      ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│  html-entities@1.2.1                       14.47 KB                  3.31%                      ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│  components                                13.18 KB                  3.02%                      ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
│                                                                                                 ││                                                                                                 │
└─────────────────────────────────────────────────────────────────────────────────────────────────┘└─────────────────────────────────────────────────────────────────────────────────────────────────┘
```

As I think that takes care of your issues, I'm going to close this issue. And you can revoke my minimal repository access now -- thanks for that, totally helped!

If you're still encountering issues related to what we've got here, feel free to reopen.

Good luck and thanks for identifying the `full` bug!

**Comment by jodriscoll** [comment]
That did it! Thanks so much! If there is any chance you could share with me how you were able to troubleshoot that issue, I'd greatly appreciate it (on your free time of course!) 🥇

**Comment by ryan-roemer** [comment]
@jodriscoll -- Sure! Diagnosis comes partially from doing a lot of work the dashboard and from the knowing the requirements for `inspectpack`: https://github.com/FormidableLabs/inspectpack#bundle , specifically the "no minification" requirement. First thing I did after getting your example up and running was to look at the actual bundle, and I saw it was minified :)

We don't really document that well _here_ in the dashboard, so I've opened #233 to have us either automagically mutate underlying configs to disable this stuff _and/or_ document things better.
